Makefile

```diff
@@ -120,18 +114,25 @@ endif
 
 .PHONY: doc-setup
 doc-setup:
+	@go version > /dev/null 2>&1 || { echo "go is not installed for lxd-doc installation."; exit 1; }
+	cd lxd/config/generate && CGO_ENABLED=0 go build -o $(GOPATH)/bin/lxd-doc
+	@echo "LXD-DOC built successfully"
 	@echo "Setting up documentation build environment"
 	python3 -m venv doc/.sphinx/venv
 	. $(SPHINXENV) ; pip install --upgrade -r doc/.sphinx/requirements.txt
 	rm -Rf doc/html
 
+.PHONY: preview-lxd-doc
```

Comment:
How about "generate-config" or something?
The generated files will also be used for the UI in the future, so not only in the docs. And there's not preview here, but a generation. ;)